### PR TITLE
Add Alien Wall

### PR DESCRIPTION
**Pull request by stefansimon76**
- Alien Wall added
- Water consumption of the steam engine reduced so that a boiler can now supply a steam engine

### CODE REVIEW

File: alien-module/prototypes/item/alien-warfare.lua [hunk 1]
```diff
@@ -62,7 +64,7 @@ if (settings.startup["alien-module-hyper-ammo-enabled"].value) then
 								},
 								{
 									type = "damage",
-									damage = { amount = 5 + 0.2 * i, type = "physical" }
+									damage = { amount = 14 + 0.2 * i, type = "physical" }
```
Comment (renoth) [inline alien-module/prototypes/item/alien-warfare.lua:67]: Should remain at 5

File: alien-module/prototypes/recipe/alien-module.lua [hunk 2]
```diff
@@ -233,7 +233,7 @@ data:extend({
 		result = "alien-fuel",
 		result_count = 1,
 		ingredients = {
-			{ "alien-plate", 2 },
+			{ "artifact-ore", 2 },
```
Comment (renoth) [inline alien-module/prototypes/recipe/alien-module.lua:236]: please revert this change

File: alien-module/prototypes/recipe/alien-warfare.lua [hunk 3]
```diff
@@ -22,7 +22,7 @@ data:extend({
 		result = "alien-ore-magazine",
 		result_count = 1,
 		ingredients = {
-			{ "artifact-ore", 2 },
+			{ "artifact-ore", 1 },
```
Comment (renoth) [inline alien-module/prototypes/recipe/alien-warfare.lua:25]: please revert

**Review by renoth — COMMENTED** [review]
Thanks for the PR, you did also change some balancing values, some of these are OK, please revert those which i commented on. Thanks!

**Comment by renoth** [comment]
One more thing i noticed, this change needs a migration so that players who already researched military also have the walls recipe enabled (see migrations/migration 1.0.9 for an example)

File: alien-module/prototypes/item/alien-warfare.lua (hunk 1 above)
Comment (stefansimon76) [reply alien-module/prototypes/item/alien-warfare.lua:67]: I think the alien ammunition (5 + 0.2 * i) is too bad compared to the mid game armor piercing ammunition (14). It should be classified between armor-piercing ammunition (14) and ammunition mixed with uranium (24).

File: alien-module/prototypes/recipe/alien-module.lua (hunk 2 above)
Comment (stefansimon76) [reply alien-module/prototypes/recipe/alien-module.lua:236]: In my opinion, it doesn't make sense to melt down alien artifacts into alien plates and then burn them. It makes more sense to process the artefact directly into fuel.

File: alien-module/prototypes/recipe/alien-warfare.lua (hunk 3 above)
Comment (stefansimon76) [reply alien-module/prototypes/recipe/alien-warfare.lua:25]: The alien-ore-magazine (4 damage) is too expensive for this short-term use in the early game.
